src/dht.js

```diff
@@ -0,0 +1,391 @@
+'use strict'
+
+const CID = require('cids')
+const PeerID = require('peer-id')
+const PeerInfo = require('peer-info')
+const multiaddr = require('multiaddr')
+const errcode = require('err-code')
+
+const {
+  Request,
+  Response,
+  DHTRequest,
+  DHTResponse
+} = require('libp2p-daemon/src/protocol')
```

Comment:
Is this whole module really a client or is it just using the daemon as an embedded node?

Reply:
This uses unix sockets for communication between processes executing on the same host (through protobuf). This way, the daemon (implementation agnostic) is a different process from this one.

More information in the [spec](https://github.com/libp2p/go-libp2p-daemon/blob/master/specs/CONTROL.md) and [roadmap](https://github.com/libp2p/go-libp2p-daemon/blob/1d7d0a25b6152441e4e8a216d53ee2f7240a161e/ROADMAP.md#introduction)

Reply:
In this case, I require the `protocol` protobuf definition from the `daemon`. 

We should probably move this definition, in the future, to an independent module and ideally have the `go` definition in the same repo.